apps/main/src/components/PageCreatePool/TokensInPool/index.tsx

```diff
@@ -154,87 +174,147 @@ const TokensInPool = ({ curve, chainId, haveSigner }: Props) => {
               address: '',
             },
           }
-        } else if (value === tokensInPool[TOKEN_B].address) {
+          // value = token in basepool that is already selected
+        } else if (
+          swapType === STABLESWAP &&
+          tokensInPoolState.tokenB.basePool &&
+          basePoolCoins.some((token) => token === value)
+        ) {
+          updatedFormValues = {
+            ...updatedFormValues,
+            [TOKEN_A]: {
+              ...updatedFormValues[TOKEN_A],
+              address: value,
+              symbol: findSymbol(value),
+            },
+            [TOKEN_B]: {
+              ...updatedFormValues[TOKEN_B],
+              address: '',
+              basePool: false,
+            },
+          }
+        } else if (value === tokensInPoolState[TOKEN_B].address) {
           swapTokens(TOKEN_A, TOKEN_B)
-        } else if (value === tokensInPool[TOKEN_C].address) {
+        } else if (value === tokensInPoolState[TOKEN_C].address) {
```

Comment:
This code seems to be repeated a lot of times.
It could also be probably replaced by a loop